hydra-node/src/Hydra/HeadLogic/State.hs

```diff
@@ -260,12 +260,14 @@ instance Arbitrary DepositStatus where
 data ClosedState tx = ClosedState
   { parameters :: HeadParameters
   , confirmedSnapshot :: ConfirmedSnapshot tx
+  , pendingDeposits :: Map (TxIdType tx) (Deposit tx)
```

Comment:
Must: not track `pendingDeposits` only in `Open` and `Closed` head states

This falls into the same pit as the original code: The list of deposits must be independent to the state of the Head (we track). Otherwise we have follow-up issues title "Recover deposits in idle state" or "Recover deposit in final state" right away.